models/things.js

```diff
@@ -28,21 +30,38 @@ var Things = {
    /**
     * Get all Things stored in the database.
     *
-    * @return Promise which resolves with a list of Thing objects.
+    * @return {Promise} which resolves with a Map of Thing objects.
     */
    getThings: function() {
      return new Promise((function(resolve, reject) {
        Database.getThings().then((function(things) {
          // Update the map of Things
-         this.things = {};
+         this.things = new Map();
```

Comment:
Interesting. I didn't realize that there was a Map and what the differences were. From my reading it looks like I should be using Map in several places in the adapter where I currently use an object {}